lib/puppet/functions/lookup.rb

```diff
@@ -0,0 +1,199 @@
+# Looks up data defined using Hiera and Data Providers
```

Comment:
Change to:
"Looks up data defined using Data Binding, and Data Providers using different strategies. The lookup searches in Data Bindings first (if configured; typically Hiera), then in the environments data provider (if any), and last in the module's data provider (if any) of the module the call to lookup originates from. Thus, the global Data Binding has higher priority than data provided in the environment, which has higher priority that data provided  in a module"